lib/ui/mockData/MockRecordAmountViewModel.dart

```diff
@@ -0,0 +1,137 @@
+import 'package:farmsmart_flutter/model/loading_status.dart';
+import 'package:farmsmart_flutter/ui/common/MockString.dart';
+import 'package:farmsmart_flutter/ui/profitloss/RecordAmount.dart';
+import 'package:farmsmart_flutter/ui/profitloss/RecordAmountHeader.dart';
+import 'package:farmsmart_flutter/ui/profitloss/RecordAmountListItem.dart';
+import 'package:flutter/material.dart';
+
+class MockRecordAmountViewModel {
+  static RecordAmountViewModel buildSale() {
+    List<RecordAmountListItemViewModel> list = [];
+
+    for (var i = 0; i < 3; i++) {
+      list.add(MockRecordAmountListItemViewModel.build(i));
+    }
+
+    return RecordAmountViewModel(
+        loadingStatus: LoadingStatus.SUCCESS,
+        actions: list,
+        buttonTitle: _mockButtonTitle[1],
+        listener: (data) => _mockTap(data),
+        type: RecordType.sale);
+  }
+
+  static RecordAmountViewModel buildCost() {
+    List<RecordAmountListItemViewModel> list = [];
+
+    for (var i = 0; i < 3; i++) {
+      list.add(MockRecordAmountListItemViewModel.build(i));
+    }
+
+    return RecordAmountViewModel(
+        loadingStatus: LoadingStatus.SUCCESS,
+        actions: list,
+        buttonTitle: _mockButtonTitle[0],
+        listener: (data) => _mockTap(data),
+        type: RecordType.cost);
+  }
+
+  static RecordAmountViewModel buildViewSale() {
+    List<RecordAmountListItemViewModel> list = [];
+
+    for (var i = 0; i < 3; i++) {
+      list.add(MockRecordAmountListItemViewModel.buildNotEditable(i));
+    }
+
+    return RecordAmountViewModel(
+      loadingStatus: LoadingStatus.SUCCESS,
+      actions: list,
+      amount: "12.48",
+      buttonTitle: _mockButtonTitle[1],
+      type: RecordType.sale,
+      isEditable: false,
+    );
+  }
+
+  static RecordAmountViewModel buildViewCost() {
+    List<RecordAmountListItemViewModel> list = [];
+
+    for (var i = 0; i < 3; i++) {
+      list.add(MockRecordAmountListItemViewModel.buildNotEditable(i));
+    }
+
+    return RecordAmountViewModel(
+        loadingStatus: LoadingStatus.SUCCESS,
+        actions: list,
+        buttonTitle: _mockButtonTitle[1],
+        type: RecordType.cost,
+        isEditable: false,
+        amount: "-233.3");
+  }
+
+  static _mockTap(RecordData data) {
+    print(data.amount);
```

Comment:
maybe print all the data not just amount

Reply:
Yes, now it's improved